app/assets/javascripts/discourse/components/reviewable-item.js.es6

```diff
@@ -0,0 +1,159 @@
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import computed from "ember-addons/ember-computed-decorators";
+import Category from "discourse/models/category";
+import optionalService from "discourse/lib/optional-service";
+
+let _components = {};
+
+export default Ember.Component.extend({
+  adminTools: optionalService(),
+  tagName: "",
+  updating: null,
+  editing: false,
+  _updates: null,
+
+  @computed("reviewable.type")
+  customClass(type) {
+    return type.dasherize();
+  },
+
+  @computed("reviewable.score")
+  displayScore(score) {
+    return score.toFixed(1);
+  },
+
+  // Find a component to render, if one exists. For example:
+  // `ReviewableUser` will return `reviewable-user`
+  @computed("reviewable.type")
+  reviewableComponent(type) {
+    if (_components[type] !== undefined) {
+      return _components[type];
+    }
+
+    let dasherized = Ember.String.dasherize(type);
+    let templatePath = `components/${dasherized}`;
+    let template = Ember.TEMPLATES[`${templatePath}`] || Ember.TEMPLATES[`javascripts/${templatePath}`];
+    if (template) {
+      _components[type] = dasherized;
+      return dasherized;
+    }
+    _components[type] = null;
+  },
+
+  _performConfirmed(action) {
+    let reviewable = this.get("reviewable");
+
+    let performAction = () => {
+      let version = reviewable.get("version");
+      this.set("updating", true);
+      return ajax(
+        `/review/${reviewable.id}/perform/${action.id}?version=${version}`,
+        {
+          method: "PUT"
+        }
+      )
+        .then(result => {
+          this.attrs.remove(
+            result.reviewable_perform_result.remove_reviewable_ids
+          );
+        })
+        .catch(popupAjaxError)
+        .finally(() => this.set("updating", false));
+    };
+
+    if (action.client_action) {
+      let actionMethod = this[`client${action.client_action.classify()}`];
+      if (actionMethod) {
+        return actionMethod.call(this, reviewable, performAction);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error(`No handler for ${action.client_action} found`);
+        return;
+      }
+      return;
+    } else {
+      return performAction();
+    }
+  },
+
+  clientSuspend(reviewable, performAction) {
+    this._penalize("showSuspendModal", reviewable, performAction);
+  },
+
+  clientSilence(reviewable, performAction) {
+    this._penalize("showSilenceModal", reviewable, performAction);
+  },
+
+  _penalize(adminToolMethod, reviewable, performAction) {
+    let adminTools = this.get("adminTools");
+    if (adminTools) {
+      let createdBy = reviewable.get("target_created_by");
```

Comment:
I wonder if `getProperties` is faster?

Reply:
I don't think it's faster when you consider it has to be de-structured too since the result is a JS object. Also it's been a while since I reviewed `getProperties` but previously it was a for loop on the properties, which is certainly slower than a few `get`s!